BaseStation/Cotroller Client Test/Program.cs

```diff
@@ -0,0 +1,54 @@
+﻿using Scarlet.Communications;
+using Scarlet.Utilities;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+// this code will go on the rover. Right now it only prints the data to a console
+
+namespace ControllerClient
+{
+    class Program
+    {
+        static string ServerIP;
+        static void Main(string[] args)
+        {
+            ServerIP = "127.0.0.1"; // ip to be changed later
+            int PortTCP = 5287;
+            int PortUDP = 5288;
+            string ClientName = "Controller Parser";
+            int recieveBufferSize = 128;
+
+            Parse.SetParseHandler(0, PrintPacketData);
+
+            Log.SetGlobalOutputLevel(Log.Severity.ERROR);
+            Client.Start(ServerIP, PortTCP, PortUDP, ClientName, recieveBufferSize);
+        }
+
+        // prints the packet raw information. In the future the data will be used to control the rover 
+        private static void PrintPacketData(Packet Packet)
```

Comment:
Turn this into a function which returns the data, rather than printing it directly. That way it is more usable when we integrate everything.

Reply:
do you want each bit of data to be sent individually or to just send the whole packet back?

Reply:
Maybe something like `ParsePacket(Packet)` which returns the information parsed in a format which is ready for use elsewhere in the code.

For testing purposes, your parse handler PrintPacketData could call ParsePacket and then print out the results.

Reply:
I'm not sure what other formats they should be sent in. should the controls be grouped based on where they're being sent to? If so i would probably want to know more about what components need which info

Reply:
For now, specify a format. It could be as simple as an object array or as complex as a `JoystickState` class. If you want you could implement this in a separate PR

Reply:
Scarlet can only send primitives through packets. Should I re-format them for other classes that will be on the same client as the receiver? If so I'm not sure how those other class files should pick up that info.

Reply:
If you want to send data other than primitives, then it would be  a good idea to serialize some data class and use that on both ends. Otherwise, your only option is to perform the serialization yourself by writing the primitives you need to a byte array which you can then decode on the other size. This is what is being done now and it works. Either option is good for our purposes. The purpose of defining a separate function is to separate the serialization (parsing) from using the data (printing).

Reply:
You can also consider using the new `SafePacketWriter`/`SafePacketScanner` functionality to make this a bit easier, but it can still only do primitives.